### PR TITLE
Introduce pad and make `Frame` immutable

### PR DESCRIPTION
**Pull request by kenoss**
fix: https://github.com/Idein/actfw-core/issues/17

Added library internal classes `_Pad*`.
`_PadDiscardingOld` is suitable for producer classes that need `Frame.update()` techniqueue.

### CODE REVIEW

**Comment by kenoss** [comment]
I checked the usage of methods of `Pipe.in_queues` and `Pipe.out_queues`. `actfw-*` is safe with this change.

On the other hand, deletion of  `Frame._update()` requires changes. Producer classes `PiCameraCapture` and `GstreamerCapture`. This follows major version updates of `actfw-*` packages.

I didn't delete `Frame` class itself because it requires updates of Actcast applications and very expensive.

File: actfw_core/util/pad.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    ) -> None:
+        # First, make the queue empty.
+        # Notice that only the owner of `_PadIn` can `put()`, but the owner can call `put()` cuncurrently.
+        i = self._queue.qsize()
+        while i != 0:
+            i -= 1
+            try:
+                self.get(block=False)
+            except Empty:
+                pass
```
Comment (notogawa) [inline actfw_core/util/pad.py:91]: Why do you use an unbounded queue as `self._queue`? `Queue(1)` makes here simple like,

```suggestion
        try:
            self.get(block=False)
        except Empty:
            pass
```

Comment (kenoss) [reply actfw_core/util/pad.py]: Nice :+1:  a45b49c

**Review by notogawa — APPROVED** [review]
LGTM

**Pull request merged**